### PR TITLE
Flow: add support for 'implies' type guard variant

### PR DESCRIPTION
**Pull request by gkz**
## Description

Adds support for Flow's `implies x is T` type guards, along with the basic one and the asserts varient. 

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/print/type-annotation.js [hunk 1]
```diff
@@ -556,7 +556,7 @@ function printTypeQuery({ node }, print) {
 function printTypePredicate(path, print) {
   const { node } = path;
   return [
-    node.asserts ? "asserts " : "",
+    node.asserts ? "asserts " : node.kind ? `${node.kind} ` : "",
```
Comment (gkz) [inline src/language-js/print/type-annotation.js:559]: Flow has replaced the `asserts` boolean flag with a `kind` field that will either by `null`, or be `"asserts"` or `"implies"`.

Comment (fisker) [reply src/language-js/print/type-annotation.js:559]: Did you mean `node.asserts` only exits in old Flow AST? Does it exists in TS?

Comment (fisker) [reply src/language-js/print/type-annotation.js:559]: Never mind, TS use that as I checked.

Comment (fisker) [reply src/language-js/print/type-annotation.js:559]: Maybe we can add `node.type` check here to make it clear.

**Review by fisker — APPROVED** [review]

**Pull request merged**